### PR TITLE
Fix usage of Process::Status on ruby3.3

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

File: ext/rubymain.cpp [hunk 1]
```diff
@@ -563,6 +563,13 @@ static VALUE t_get_subprocess_status (VALUE self UNUSED, VALUE signature)
 	if (evma_get_subprocess_status (NUM2BSIG (signature), &status)) {
 		if (evma_get_subprocess_pid (NUM2BSIG (signature), &pid)) {
 
+#ifdef IS_RUBY_3_3_OR_LATER
```
Comment (sodabrew) [inline ext/rubymain.cpp:566]:
```suggestion
#if defined(IS_RUBY_3_3_OR_LATER)
```

File: ext/rubymain.cpp [hunk 2]
```diff
@@ -563,6 +563,13 @@ static VALUE t_get_subprocess_status (VALUE self UNUSED, VALUE signature)
...
+			proc_status = rb_obj_alloc(rb_cProcessStatus);
+			struct rb_process_status *data = NULL;
+			data = (rb_process_status*)RTYPEDDATA_GET_DATA(proc_status);
+			data->pid = pid;
+			data->status = status;
+#else
 #ifdef IS_RUBY_3_OR_LATER
```
Comment (sodabrew) [inline ext/rubymain.cpp:573]:
```suggestion
#elif defined(IS_RUBY_3_OR_LATER)
```

File: ext/rubymain.cpp [hunk 3]
```diff
@@ -585,6 +592,7 @@ static VALUE t_get_subprocess_status (VALUE self UNUSED, VALUE signature)
 			rb_ivar_set(proc_status, rb_intern_const("status"), INT2FIX(status));
 			rb_ivar_set(proc_status, rb_intern_const("pid"), INT2FIX(pid));
 #endif
+#endif
```
Comment (sodabrew) [inline ext/rubymain.cpp:595]:
```suggestion
```

**Comment by sodabrew** [comment]
Thank you! I think this will be a bit clearer with `#if defined()`

**Comment by terceiro** [comment]
@sodabrew updated, thanks for the review

**Pull request merged**